samples/net/lwm2m_client/README.rst

```diff
@@ -6,14 +6,144 @@ LWM2M client
 Overview
 ********
 
-TODO
+Lightweight Machine to Machine (LwM2M) is an application layer protocol
+based on CoAP/UDP. and is designed to expose various resources for reading,
+writing and executing via an LwM2M server in a very lightweight environment.
+
+This LwM2M client sample application for Zephyr implements the LwM2M library
+and establishes both IPv4 and IPv6 connections to a LwM2M server using
+the Open Mobile Alliance "Client Registration Interface".
+
+The source code for this sample application can be found at:
+:file:`samples/net/lwm2m_client`.
+
+Requirements
+************
+
+- :ref:`networking_with_qemu`
+- Linux machine
+- Leshan Demo server (source: https://github.com/eclipse/leshan)
```

Comment:
Oh.  In that case it's fine (I'm not an eclipse user :)  Use the most appropriate link that will get you to the demo server.